presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -1450,36 +1569,40 @@ public void testBaseToViewConversionWithDerivedFields()
                 .setSystemProperty(QUERY_OPTIMIZATION_WITH_MATERIALIZED_VIEW_ENABLED, "true")
                 .build();
         QueryRunner queryRunner = getQueryRunner();
-        String baseTable = "lineitem_partitioned_derived_fields";
+        String table = "lineitem_partitioned_derived_fields";
         String view = "lineitem_partitioned_view_derived_fields";
         try {
             queryRunner.execute(format(
                     "CREATE TABLE %s WITH (partitioned_by = ARRAY['ds', 'shipmode']) AS " +
                     "SELECT discount, extendedprice, '2020-01-01' as ds, shipmode FROM lineitem WHERE orderkey < 1000 " +
                     "UNION ALL " +
                     "SELECT discount, extendedprice, '2020-01-02' as ds, shipmode FROM lineitem WHERE orderkey > 1000",
-                    baseTable));
+                    table));
             assertUpdate(format(
                     "CREATE MATERIALIZED VIEW %s WITH (partitioned_by = ARRAY['mvds', 'shipmode']) AS " +
                             "SELECT SUM(discount * extendedprice) as _discount_multi_extendedprice_ , MAX(discount*extendedprice) as _max_discount_multi_extendedprice_ , " +
                             "ds as mvds, shipmode FROM %s group by ds, shipmode",
-                    view,
-                    baseTable));
+                    view, table));
             assertTrue(getQueryRunner().tableExists(getSession(), view));
             ExtendedHiveMetastore metastore = replicateHiveMetastore((DistributedQueryRunner) queryRunner);
-            appendTableParameter(metastore, baseTable, REFERENCED_MATERIALIZED_VIEWS, format("%s.%s", getSession().getSchema().orElse(""), view));
+            appendTableParameter(metastore, table, REFERENCED_MATERIALIZED_VIEWS, format("%s.%s", getSession().getSchema().orElse(""), view));
             assertUpdate(format("REFRESH MATERIALIZED VIEW %s where mvds='2020-01-01'", view), 7);
             String baseQuery = format(
                     "SELECT sum(discount * extendedprice) as _discount_multi_extendedprice_ , MAX(discount*extendedprice) as _max_discount_multi_extendedprice_ , " +
-                            "ds, shipmode as method from %s group by ds, shipmode ORDER BY ds, shipmode",
-                    baseTable);
+                            "ds, shipmode as method from %s group by ds, shipmode ORDER BY ds, shipmode", table);
             MaterializedResult optimizedQueryResult = computeActual(queryOptimizationWithMaterializedView, baseQuery);
             MaterializedResult baseQueryResult = computeActual(baseQuery);
             assertEquals(optimizedQueryResult, baseQueryResult);
+
+            assertPlan(queryOptimizationWithMaterializedView, baseQuery, anyTree(
```

Comment:
may we can also check here that the plan generated for baseQuery and viewQuery should be same.